Java/jni/native_view.cc

```diff
@@ -149,24 +149,26 @@ JNIEXPORT void JNICALL Java_com_couchbase_cbforest_View_emit(JNIEnv *env, jobjec
     jlong *keys   = env->GetLongArrayElements(jkeys, NULL);
     C4Key *c4keys[count];
     C4Slice c4values[count];
-    std::vector<jbyteArraySlice> valueBufs;
+    std::vector<jbyteArraySlice*> valueBufs;
     for(int i = 0; i < count; i++) {
         c4keys[i] = (C4Key*)keys[i];
         jbyteArray jvalue = (jbyteArray) env->GetObjectArrayElement(jvalues, i);
         if (jvalue) {
-            valueBufs.push_back(jbyteArraySlice(env, jvalue));
```

Comment:
Hi @snej,

I tried `std::vector.emplace`, simply replaced `push_back` with `emplace`.
```
// valueBufs.push_back(jbyteArraySlice(env, jvalue));
valueBufs.emplace(valueBufs.end(), jbyteArraySlice(env, jvalue));
```

However, it still causes same problem `push_back` caused.

Error message
```
java(22084,0x11fdbf000) malloc: *** error for object 0x7fccb86973a0: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```